### PR TITLE
Create average.c

### PR DESCRIPTION
**Pull request by lukasb1b**
adding an average function with a test case

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: math/average.c [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    
+    while(avg[size+1]!='\0'){
+        size++;
+    }
+
+    for (int i = 0; i < size; i++) {
+        result += avg[i];
+    }
+
+    return result / size;
```
Comment (alexpantyukhin) [inline math/average.c:18]: what if the size is zero?

File: math/average.c [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#include <stdio.h>
+#include <assert.h>
+
+//average of an array
+
+double average(int avg[]) {
+    double result = 0;
+    int size=0;
+    
+    while(avg[size+1]!='\0'){
```
Comment (alexpantyukhin) [inline math/average.c:10]: I think this case you can find the array size without a loop. For example: `sizeof(avg) / sizeof(int)` What do you think?

File: math/average.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#include <stdio.h>
+#include <assert.h>
+
+//average of an array
+
+double average(int avg[]) {
```
Comment (alexpantyukhin) [inline math/average.c:6]: why avg? is it the array of avgs? Maybe values? What do you think?

File: math/average.c (hunk 1 above)
Comment (lukasb1b) [reply math/average.c:18]: lets do something like if (size== 0) {
        perror("empty array");
    }

File: math/average.c (hunk 2 above)
Comment (lukasb1b) [reply math/average.c:10]: I tried it too but i did not get it to work

File: math/average.c (hunk 3 above)
Comment (lukasb1b) [reply math/average.c:6]: can you explain this please

File: math/average.c (hunk 2 above)
Comment (CascadingCascade) [reply math/average.c:10]: IIRC, `sizeof(avg) / sizeof(int)` won't work here, an array degrades into a pointer to its first element when passed into a function, and dividing the size of a pointer obviously does nothing

Comment (alexpantyukhin) [reply math/average.c:10]: yes, you are right.

File: math/average.c (hunk 3 above)
Comment (rbevin777) [inline math/average.c:6]: Don't forget to add the function header. This should look something like this
```
/**
 * @brief    This is a brief description of what the function does
 * @param param_name - This is a brief description of the parameter
 * @return return_name - Biref description of the return value itself.
 */
```

File: math/average.c (hunk 1 above)
Comment (rbevin777) [inline math/average.c:18]: It'd be useful to declare a variable like this so that if you or someone else needs to debug the code it is eaiser to trace.
```
int avg_val = result / size;
return avg_val;
```

File: math/average.c [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    for (int i = 0; i < size; i++) {
+        result += avg[i];
+    }
+
+    return result / size;
+}
+
+int main() {
+    int test[] = {1,2,3,4,5};
```
Comment (rbevin777) [inline math/average.c:22]: You could add a few different tests to ensure the robustness, so for example using all negative numbers, a mix of negative and positive, you could go for large numbers close to the maximum size of an `int` and see what happens there (most likely an overflow and the wrong result) which is what you want to see from tests anyway. Play about with testing. It's great at highlighting the faults.

Comment (rbevin777) [inline math/average.c:22]: Another comment in regards to testing, there is a common testing documentation format called [gherkin](https://cucumber.io/docs/gherkin/reference/) The basics of it are 

```
// Given we have an array fo values

// When we try to work out the average value of the array

// Then we should get the expected result
```

This is a simple example but simple is good.

**Review by rbevin777 — CHANGES_REQUESTED** [review]
Good module, also don't be disheartened by the number of comments either! 😁

**Comment by lukasb1b** [comment]
I changed all of your requests exept of the last one and the last test is not passing but i dont know why because it prints15.000000-15.000000 and I think that should be 0 any fixes

**Comment by lukasb1b** [comment]
I made a workaround at the tests by setting the last value 0 but otherwise i don't think there is a solution to calulate the array length. I just can think of creating a parameter for the length

**Comment by Adlinke** [comment]
I am so excited to having you! Thank you so much!
